src/bun.js/bindings/v8/V8Object.cpp

```diff
@@ -62,10 +62,10 @@ Maybe<bool> Object::Set(Local<Context> context, Local<Value> key, Local<Value> v
 
 void Object::SetInternalField(int index, Local<Data> data)
 {
-    auto fields = getInternalFieldsContainer(this);
-    if (fields && index >= 0 && index < fields->size()) {
-        fields->at(index) = InternalFieldObject::InternalField(data->localToJSValue(Isolate::GetCurrent()->globalInternals()));
-    }
+    auto* fields = getInternalFieldsContainer(this);
+    RELEASE_ASSERT(fields, "object has no internal fields");
+    RELEASE_ASSERT(index >= 0 && index < fields->size(), "internal field index is out of bounds");
+    fields->at(index) = data->localToJSValue(Isolate::GetCurrent()->globalInternals());
```

Comment:
Do we need a lock for this?

Reply:
I don't think so. [It doesn't seem like V8's API is thread-safe either](https://stackoverflow.com/a/7921673).